src/github_utils.py

```diff
@@ -19,48 +19,56 @@ def get_pull_request_diff():
     # Get diff content
     diff_url = pull_request.diff_url
     response = requests.get(diff_url)
-    print(response.text)
+    #print(response.text)
     return response.text
 
 
-def get_valid_lines(diff_content):
-    """Parse diff to find valid line numbers with proper initialization"""
-    valid_lines = {}
+def get_valid_hunks(diff_content):
+    """Parse diff to find valid hunk positions"""
+    hunks = []
+    current_hunk = None
     current_file = None
-    line_number = None
     
     for line in diff_content.split('\n'):
-        # Reset tracking for new files
         if line.startswith('diff --git'):
-            current_file = line.split(' b/')[-1].strip()
-            valid_lines[current_file] = set()
-            line_number = None  # Reset for new file
-            continue
+            # New file section
+            current_file = line.split(' b/')[1].split()[0]
+        elif line.startswith('@@'):
+            # Hunk header format: @@ -old_start,old_lines +new_start,new_lines @@
+            parts = line.split(' ')
+            new_part = parts[2].split(',')
+            new_start = int(new_part[0][1:])
+            hunk_length = int(new_part[1])
             
-        # Parse hunk header
-        if line.startswith('@@'):
-            try:
-                # Extract new file line number (after +)
-                hunk_header = line.split('+', 1)[1].split(' ', 1)[0]
-                line_number = int(hunk_header.split(',')[0])
-                valid_lines[current_file].add(line_number)
-            except (IndexError, ValueError):
-                line_number = None  # Invalid hunk format
-            continue
-            
-        # Only process lines if we have valid line_number
-        if line_number is None:
-            continue
-            
-        # Track added lines
```

Comment:
This function's name is misleading. It doesn't just validate, but also transforms the comment. Consider renaming it to `transform_to_github_comment`.

```suggestion
transform_to_github_comment
```